### PR TITLE
removes empty AddRound Form when click outside

### PR DESCRIPTION
**Pull request by jbrambleDC**
My first time using react hooks ...... teat it apart 😉

### CODE REVIEW

File: app/src/lib/click.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from "react"
+
+export function useOnClickOutside(ref: any, handler: CallableFunction) {
```
Comment (jbrambleDC) [inline app/src/lib/click.ts:3]: I should probably not use any type here?

File: app/src/pages/Lobby/ControllableRoundSettings.tsx [hunk 2]
```diff
@@ -30,55 +31,98 @@ function ControllableRoundSettings() {
   const [addRound] = useAddRoundMutation()
   const [showAddRoundForm, setShowAddRoundForm] = React.useState(false)
   const [addRoundValue, setAddRoundValue] = React.useState("")
+  const ref = React.useRef<HTMLDivElement>(null);
+
+  useOnClickOutside(ref, () => {
+    if (showAddRoundForm && addRoundValue.length === 0) {
+      setShowAddRoundForm(false)
+      setAddRoundValue("")
```
Comment (jbrambleDC) [inline app/src/pages/Lobby/ControllableRoundSettings.tsx:39]: This is not needed

File: app/src/lib/click.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from "react"
```
Comment (avimoondra) [inline app/src/lib/click.ts:1]: would rename this file `useOnClickOutside` like useInterval, and maybe leave a comment w/this reference code - https://usehooks.com/useOnClickOutside/

and newline at EOF

File: app/src/lib/click.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+export function useOnClickOutside(ref: React.RefObject<any>, handler: CallableFunction) {
```
Comment (avimoondra) [inline app/src/lib/click.ts:3]: any seems ok here. CallableFunction is new too me, looks like it works - I usually use `(event: MouseEvent | TouchEvent) => void`

File: app/src/lib/click.ts (hunk 3 above)
Comment (jbrambleDC) [reply app/src/lib/click.ts]: Done

**Pull request merged**